implementation/src/main/java/io/smallrye/mutiny/Uni.java

```diff
@@ -382,12 +383,20 @@ static UniCombine combine() {
 
     /**
      * Caches the events (item or failure) of this {@link Uni} and replays it for all further {@link UniSubscriber}.
+     * This method is equivalent to {@code uni.cacheItemOrFailure().indefinitely()}.
      *
      * @return the new {@link Uni}. Unlike regular {@link Uni}, re-subscribing to this {@link Uni} does not re-compute
      *         the outcome but replayed the cached events.
      */
     Uni<T> cache();
 
+    /**
+     * Configures cache behaviour of this {@link Uni}.
+     * 
+     * @return a {@link UniCacheItemOrFailure} to configure cache behaviour.
+     */
+    UniCacheItemOrFailure<T> cacheItemOrFailure();
```

Comment:
Had no real good idea for the naming, better names are welcome.

Furthermore: I assome you want some words about this in the documentation? I would provide this as soon we are clear about the the implementation.